src/main/java/org/sagebionetworks/bridge/services/ParticipantService.java

```diff
@@ -846,9 +862,10 @@ public StudyParticipant updateIdentifiers(App app, CriteriaContext context, Iden
         return getParticipant(app, account.getId(), false);
     }
 
-    public void downloadParticipantRoster(String appId, String userId, String password, String studyId) throws JsonProcessingException {
-        // password must contain at least 8 characters, 1 upper case, 1 lower case, and 1 number
-        checkPassword(password);
+    public void downloadParticipantRoster(String appId, String userId, String password, String studyId,
+                                          ParticipantRosterRequest request) throws JsonProcessingException {
+        validatePassword(password);
+        Validate.entityThrowingException(new ParticipantRosterRequestValidator(), request);
```

Comment:
We can create a static instance ParticipantRosterRequestValidator.INSTANCE so we don't have to create a new one with every request.

Reply:
wow interesting I learned something new!